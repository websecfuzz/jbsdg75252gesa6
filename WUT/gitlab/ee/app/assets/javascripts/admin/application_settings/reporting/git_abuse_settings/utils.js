import { parseBoolean } from '~/lib/utils/common_utils';

export const parseFormProps = ({
  maxNumberOfRepositoryDownloads,
  maxNumberOfRepositoryDownloadsWithinTimePeriod,
  gitRateLimitUsersAllowlist,
  gitRateLimitUsersAlertlist,
  autoBanUserOnExcessiveProjectsDownload,
}) => ({
  maxNumberOfRepositoryDownloads: parseInt(maxNumberOfRepositoryDownloads, 10),
  maxNumberOfRepositoryDownloadsWithinTimePeriod: parseInt(
    maxNumberOfRepositoryDownloadsWithinTimePeriod,
    10,
  ),
  gitRateLimitUsersAllowlist: JSON.parse(gitRateLimitUsersAllowlist),
  gitRateLimitUsersAlertlist: JSON.parse(gitRateLimitUsersAlertlist),
  autoBanUserOnExcessiveProjectsDownload: parseBoolean(autoBanUserOnExcessiveProjectsDownload),
});
