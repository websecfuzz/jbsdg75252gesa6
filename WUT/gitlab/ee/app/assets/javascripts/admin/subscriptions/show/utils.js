import {
  subscriptionTypes,
  offlineCloudLicenseText,
  onlineCloudLicenseText,
  licenseFileText,
} from './constants';

export const getLicenseFromData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.license;
export const getErrorsAsData = ({ data } = {}) => data?.gitlabSubscriptionActivate?.errors || [];

export function getLicenseTypeLabel(type) {
  switch (type) {
    case subscriptionTypes.OFFLINE_CLOUD:
      return offlineCloudLicenseText;
    case subscriptionTypes.ONLINE_CLOUD:
      return onlineCloudLicenseText;
    default:
      return licenseFileText;
  }
}
