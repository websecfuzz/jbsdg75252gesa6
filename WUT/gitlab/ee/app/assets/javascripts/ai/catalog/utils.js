import { formValidators } from '@gitlab/ui/dist/utils';
import { s__, sprintf } from '~/locale';

export const createFieldValidators = ({ requiredLabel, maxLength } = {}) => {
  const validators = [];

  if (requiredLabel !== undefined) {
    validators.push(formValidators.required(requiredLabel));
  }

  if (maxLength !== undefined) {
    validators.push(
      formValidators.factory(
        sprintf(
          s__('AICatalog|Input cannot exceed %{value} characters. Please shorten your input.'),
          {
            value: maxLength,
          },
        ),
        (value) => (value?.length || 0) <= maxLength,
      ),
    );
  }

  return validators;
};
