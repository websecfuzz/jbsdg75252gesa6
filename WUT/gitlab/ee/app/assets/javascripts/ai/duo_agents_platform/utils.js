import { s__ } from '~/locale';
import { humanize } from '~/lib/utils/text_utility';

export const formatAgentDefinition = (agentDefinition) => {
  return humanize(agentDefinition || s__('DuoAgentsPlatform|Agent session'));
};

export const formatAgentFlowName = (agentDefinition, id) => {
  return `${formatAgentDefinition(agentDefinition)} #${id}`;
};

export const formatAgentStatus = (status) => {
  return status ? humanize(status.toLowerCase()) : s__('DuoAgentsPlatform|Unknown');
};
