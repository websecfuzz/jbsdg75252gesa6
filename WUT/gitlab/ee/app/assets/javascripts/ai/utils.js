import { duoChatGlobalState } from '~/super_sidebar/constants';
import { setCookie } from '~/lib/utils/common_utils';
import {
  DUO_AGENTIC_MODE_COOKIE,
  DUO_AGENTIC_MODE_COOKIE_EXPIRATION,
} from 'ee/ai/tanuki_bot/constants';

export const concatStreamedChunks = (arr) => {
  if (!arr) return '';

  let end = arr.findIndex((el) => !el);

  if (end < 0) end = arr.length;

  return arr.slice(0, end).join('');
};

/**
 * setCookie wrapper with duo agentic mode constances.
 *
 * @param {isAgenticMode} Boolean - Value to save
 * @returns {void}
 */
export const saveDuoAgenticModePreference = (isAgenticMode) => {
  setCookie(DUO_AGENTIC_MODE_COOKIE, isAgenticMode, {
    expires: DUO_AGENTIC_MODE_COOKIE_EXPIRATION,
  });
};

/**
 * Swith duo chat based on agenticMode value and save to cookie based on
 * saveCookie value.
 *
 * @param {agenticMode} Boolean - Agentic mode state
 * @param {saveCookie} Boolean - Save to cookie flag
 * @returns {void}
 */

export const setAgenticMode = (agenticMode = true, saveCookie = false) => {
  duoChatGlobalState.isShown = !agenticMode;
  duoChatGlobalState.isAgenticChatShown = agenticMode;

  if (saveCookie) {
    saveDuoAgenticModePreference(agenticMode);
  }
};

/**
 * Sends a command to DuoChat to execute on. This should be use for
 * a single command.
 *
 * @param {question} String - Prompt to send to the chat endpoint
 * @param {resourceId} String - Unique ID to bind the streaming
 * @param {variables} Object - Additional variables to pass to graphql chat mutation
 */
export const sendDuoChatCommand = ({ question, resourceId, variables = {} } = {}) => {
  if (!question || !resourceId) {
    throw new Error('Both arguments `question` and `resourceId` are required');
  }
  setAgenticMode(false, true);

  window.requestIdleCallback(() => {
    duoChatGlobalState.commands.push({
      question,
      resourceId,
      variables,
    });
  });
};

export const clearDuoChatCommands = () => {
  duoChatGlobalState.commands = [];
};

/**
 * Converts a text string into a URL-friendly format for event tracking.
 *
 * - Converts to lowercase
 * - Removes special characters
 * - Replaces spaces with underscores
 * - Limits length to 50 characters
 *
 * @param {string} text - The text to convert
 * @returns {string} The formatted event label
 */
export const generateEventLabelFromText = (text) => {
  return text
    .toLowerCase()
    .replace(/[^a-z0-9\s]/g, '')
    .replace(/\s+/g, '_')
    .substring(0, 50);
};

export const utils = {
  concatStreamedChunks,
  generateEventLabelFromText,
};
