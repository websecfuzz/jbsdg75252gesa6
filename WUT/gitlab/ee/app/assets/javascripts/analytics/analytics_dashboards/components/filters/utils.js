import { queryToObject } from '~/lib/utils/url_utility';
import { formatDate, newDate } from '~/lib/utils/datetime_utility';
import { ISO_SHORT_FORMAT } from '~/vue_shared/constants';
import {
  convertObjectPropsToCamelCase,
  convertObjectPropsToSnakeCase,
  parseBoolean,
} from '~/lib/utils/common_utils';
import {
  filterToQueryObject,
  urlQueryToFilter,
} from '~/vue_shared/components/filtered_search_bar/filtered_search_utils';
import {
  START_DATES,
  DATE_RANGE_OPTIONS,
  DATE_RANGE_OPTION_CUSTOM,
  DATE_RANGE_OPTION_KEYS,
  DEFAULT_SELECTED_DATE_RANGE_OPTION,
  FILTERED_SEARCH_SUPPORTED_TOKENS,
  PROJECT_FILTER_QUERY_NAME,
} from './constants';

const isCustomOption = (option) => option && option === DATE_RANGE_OPTION_CUSTOM;

export const getDateRangeOption = (optionKey) => DATE_RANGE_OPTIONS[optionKey] || null;

export const dateRangeOptionToFilter = ({ startDate, endDate, key }) => ({
  startDate,
  endDate,
  dateRangeOption: key,
});

const DEFAULT_FILTER = dateRangeOptionToFilter(
  DATE_RANGE_OPTIONS[DEFAULT_SELECTED_DATE_RANGE_OPTION],
);

export const buildDefaultDashboardFilters = (queryString, dashboardDefaultFilters = {}) => {
  const {
    dateRangeOption,
    startDate,
    endDate,
    filterAnonUsers,
    [PROJECT_FILTER_QUERY_NAME]: projectFullPath,
  } = convertObjectPropsToCamelCase(queryToObject(queryString, { gatherArrays: true }));
  const searchFilters = urlQueryToFilter(queryString, {
    filterNamesAllowList: FILTERED_SEARCH_SUPPORTED_TOKENS,
  });

  const isDateRangeFilterEnabled = dashboardDefaultFilters?.dateRange?.enabled;
  const optionKey = dateRangeOption || dashboardDefaultFilters?.dateRange?.defaultOption;
  const dateRangeOverride = DATE_RANGE_OPTION_KEYS.includes(optionKey)
    ? dateRangeOptionToFilter(getDateRangeOption(optionKey))
    : {};

  const customDateRange = isCustomOption(optionKey);

  return {
    ...(isDateRangeFilterEnabled && {
      ...DEFAULT_FILTER,
      // Override default filter with user defined option
      ...dateRangeOverride,
      // Override date range when selected option is custom date range
      ...(customDateRange && { startDate: newDate(startDate) }),
      ...(customDateRange && { endDate: newDate(endDate) }),
    }),
    filterAnonUsers: parseBoolean(filterAnonUsers),
    searchFilters,
    projectFullPath: projectFullPath || null,
  };
};

export const filtersToQueryParams = ({
  dateRangeOption,
  startDate,
  endDate,
  filterAnonUsers,
  searchFilters,
  projectFullPath,
}) => {
  const customDateRange = isCustomOption(dateRangeOption);

  const searchFiltersQueryObj = filterToQueryObject(searchFilters);
  const additionalFiltersQueryObj = convertObjectPropsToSnakeCase({
    dateRangeOption,
    // Clear the date range unless the custom date range is selected
    startDate: customDateRange ? formatDate(startDate, ISO_SHORT_FORMAT) : null,
    endDate: customDateRange ? formatDate(endDate, ISO_SHORT_FORMAT) : null,
    // Clear the anon users filter unless truthy
    filterAnonUsers: filterAnonUsers || null,
    [PROJECT_FILTER_QUERY_NAME]: projectFullPath || null,
  });

  return { ...additionalFiltersQueryObj, ...searchFiltersQueryObj };
};

export function isDashboardFilterEnabled(filter) {
  return filter?.enabled || false;
}

export const getStartDate = (option) =>
  START_DATES[option] ?? START_DATES[DEFAULT_SELECTED_DATE_RANGE_OPTION];
