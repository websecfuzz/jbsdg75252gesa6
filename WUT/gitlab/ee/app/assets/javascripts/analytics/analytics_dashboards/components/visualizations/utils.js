import { isNil } from 'lodash';
import { isNumeric } from '~/lib/utils/number_utils';
import { formatNumber, n__, __, sprintf } from '~/locale';
import { formatDate, humanizeTimeInterval } from '~/lib/utils/datetime/date_format_utility';
import { formatAsPercentage } from 'ee/analytics/dora/components/util';
import { NULL_SERIES_ID } from 'ee/analytics/shared/constants';
import { UNITS } from '~/analytics/shared/constants';

function isIsoDateString(dateString) {
  // Matches an ISO date string in the format `2024-03-14T00:00:00.000`
  const isoDateRegex = /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}$/;
  return isoDateRegex.test(dateString);
}

export function formatVisualizationValue(value) {
  if (isIsoDateString(value)) {
    return formatDate(value);
  }

  if (isNumeric(value)) {
    return formatNumber(parseInt(value, 10));
  }

  return value;
}

export function formatVisualizationTooltipTitle(title, params) {
  const value = params?.seriesData?.at(0)?.value?.at(0);

  if (isIsoDateString(value)) {
    const formattedDate = formatDate(value);
    return title.replace(value, formattedDate);
  }

  return title;
}

export function customFormatVisualizationTooltipTitle(params, formatter) {
  const xAxisValue = params?.seriesData?.at(0)?.value?.at(0);

  if (isNil(xAxisValue)) return '';

  return formatter(xAxisValue);
}

export const humanizeDisplayUnit = ({ unit, data = 0 }) => {
  switch (unit) {
    case 'days':
      return n__('day', 'days', data === '-' ? 0 : data);
    case 'per_day':
      return __('/day');
    case 'percent':
      return '%';
    default:
      return unit;
  }
};

/**
 * Humanizes values to be displayed in chart tooltips
 *
 * @param {string} unit â€“ The unit of measurement to be used for metric
 * @param {number} value - The value of the metric
 * @returns {string|number} - Humanized tooltip value
 */
export const humanizeChartTooltipValue = ({ unit, value } = {}) => {
  if (isNil(value)) return __('No data');

  switch (unit) {
    case UNITS.DAYS:
      return n__('%d day', '%d days', value);
    case UNITS.PER_DAY:
      return sprintf(__('%{value} /day'), { value });
    case UNITS.PERCENT:
      return formatAsPercentage(value);
    case UNITS.TIME_INTERVAL:
      return humanizeTimeInterval(value);
    default:
      return value;
  }
};

export const calculateDecimalPlaces = ({ data, decimalPlaces } = {}) => {
  return (data && parseInt(decimalPlaces, 10)) || 0;
};

export const removeNullSeries = (seriesData) => {
  return seriesData?.filter(({ seriesId }) => seriesId !== NULL_SERIES_ID);
};
