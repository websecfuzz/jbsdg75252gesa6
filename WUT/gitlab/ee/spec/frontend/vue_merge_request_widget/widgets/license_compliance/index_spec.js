import MockAdapter from 'axios-mock-adapter';
import { mountExtended } from 'helpers/vue_test_utils_helper';
import waitForPromises from 'helpers/wait_for_promises';
import api from '~/api';
import axios from '~/lib/utils/axios_utils';
import ActionButtons from '~/vue_merge_request_widget/components/widget/action_buttons.vue';
import Widget from '~/vue_merge_request_widget/components/widget/widget.vue';

import licenseComplianceExtension from 'ee/vue_merge_request_widget/widgets/license_compliance/index.vue';
import { HTTP_STATUS_INTERNAL_SERVER_ERROR, HTTP_STATUS_OK } from '~/lib/utils/http_status';
import {
  licenseComplianceNewLicenses,
  licenseComplianceSuccessExpanded,
  licenseComplianceRemovedLicenses,
  licenseComplianceNewAndRemovedLicenses,
  licenseComplianceNewDeniedLicenses,
  licenseComplianceNewDeniedLicensesAndExisting,
  licenseComplianceNewAndRemovedLicensesApprovalRequired,
  licenseComplianceNewDeniedLicensesAndExistingApprovalRequired,
  licenseComplianceEmpty,
  licenseComplianceEmptyExistingLicense,
  licenseComplianceExistingAndNewLicenses,
} from './mock_data';

describe('License Compliance extension', () => {
  let wrapper;
  let mock;

  const licenseComparisonPath =
    '/group-name/project-name/-/merge_requests/78/license_scanning_reports';
  const licenseComparisonPathCollapsed =
    '/group-name/project-name/-/merge_requests/78/license_scanning_reports_collapsed';
  const fullReportPath = '/group-name/project-name/-/merge_requests/78/full_report';
  const settingsPath = '/group-name/project-name/-/licenses#licenses';
  const apiApprovalsPath = '/group-name/project-name/-/licenses#policies';

  const mockApi = (endpoint, statusCode, data) => {
    mock.onGet(endpoint).reply(statusCode, data, {});
  };

  const findWidget = () => wrapper.findComponent(Widget);
  const findToggleCollapsedButton = () => wrapper.findByTestId('toggle-button');
  const findAllExtensionListItems = () => wrapper.findAllByTestId('extension-list-item');
  const findActionButtons = () => wrapper.findComponent(ActionButtons);
  const findByHrefAttribute = (href) => wrapper.find(`[href="${href}"]`);
  const findFullReportLink = () => findByHrefAttribute(fullReportPath);
  const findSummary = () => wrapper.findByTestId('widget-extension-top-level-summary');

  const createComponent = (licenseComplianceProps = {}) => {
    wrapper = mountExtended(licenseComplianceExtension, {
      propsData: {
        mr: {
          licenseCompliance: {
            license_scanning_comparison_path: licenseComparisonPath,
            license_scanning_comparison_collapsed_path: licenseComparisonPathCollapsed,
            api_approvals_path: apiApprovalsPath,
            license_scanning: {
              settings_path: settingsPath,
              full_report_path: fullReportPath,
            },
            ...licenseComplianceProps,
          },
        },
      },
    });
  };

  beforeEach(() => {
    jest.spyOn(api, 'trackRedisCounterEvent').mockImplementation(() => {});
    mock = new MockAdapter(axios);
  });

  afterEach(() => {
    mock.restore();
  });

  it('emits loaded event', async () => {
    mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

    createComponent();

    await waitForPromises();

    expect(wrapper.emitted('loaded')[0]).toContain(4);
  });

  describe('summary', () => {
    it('displays loading text', () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

      createComponent();

      expect(findSummary().text()).toBe('License Compliance test metrics results are being parsed');
    });

    it('displays failed loading text', async () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_INTERNAL_SERVER_ERROR);

      createComponent();

      await waitForPromises();
      expect(findSummary().text()).toBe('License Compliance failed loading results');
    });

    it.each`
      scenario                                                            | response                                                         | isExpandable | message
      ${'licenseComplianceEmpty'}                                         | ${licenseComplianceEmpty}                                        | ${false}     | ${'License Compliance detected no licenses for the source branch only'}
      ${'licenseComplianceEmptyExistingLicense'}                          | ${licenseComplianceEmptyExistingLicense}                         | ${false}     | ${'License Compliance detected no new licenses'}
      ${'licenseComplianceNewLicenses'}                                   | ${licenseComplianceNewLicenses}                                  | ${true}      | ${'License Compliance detected 4 licenses for the source branch only'}
      ${'licenseComplianceRemovedLicenses'}                               | ${licenseComplianceRemovedLicenses}                              | ${false}     | ${'License Compliance detected no licenses for the source branch only'}
      ${'licenseComplianceNewAndRemovedLicenses'}                         | ${licenseComplianceNewAndRemovedLicenses}                        | ${true}      | ${'License Compliance detected 2 licenses for the source branch only'}
      ${'licenseComplianceNewAndRemovedLicensesApprovalRequired'}         | ${licenseComplianceNewAndRemovedLicensesApprovalRequired}        | ${true}      | ${'License Compliance detected 4 licenses and policy violations for the source branch only; approval required'}
      ${'licenseComplianceNewDeniedLicenses'}                             | ${licenseComplianceNewDeniedLicenses}                            | ${true}      | ${'License Compliance detected 4 licenses and policy violations for the source branch only'}
      ${'licenseComplianceNewDeniedLicensesAndExisting'}                  | ${licenseComplianceNewDeniedLicensesAndExisting}                 | ${true}      | ${'License Compliance detected 4 new licenses and policy violations'}
      ${'licenseComplianceNewDeniedLicensesAndExistingApprovalRequired '} | ${licenseComplianceNewDeniedLicensesAndExistingApprovalRequired} | ${true}      | ${'License Compliance detected 4 new licenses and policy violations; approval required'}
      ${'licenseComplianceExistingAndNewLicenses'}                        | ${licenseComplianceExistingAndNewLicenses}                       | ${true}      | ${'License Compliance detected 6 new licenses'}
    `(
      'the $scenario scenario expects the message to be "$message"',
      async ({ response, message, isExpandable }) => {
        mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, response);
        createComponent();

        await waitForPromises();

        expect(findSummary().text()).toBe(message);
        expect(findToggleCollapsedButton().exists()).toBe(isExpandable);
      },
    );

    it('displays the help popover', () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

      createComponent();

      expect(findWidget().props('helpPopover')).toEqual({
        content: {
          learnMorePath:
            '/help/user/compliance/license_approval_policies#criteria-to-compare-licenses-detected-in-the-merge-request-branch-to-licenses-in-the-default-branch',
          text: 'Detects known vulnerabilities in your software dependencies.',
        },
        options: {
          title: 'License scan results',
        },
      });
    });
  });

  describe('actions buttons', () => {
    it('displays manage licenses and full report links', async () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

      createComponent();

      await waitForPromises();

      expect(findFullReportLink().exists()).toBe(true);
      expect(findFullReportLink().text()).toBe('Full report');

      expect(findActionButtons().exists()).toBe(true);
    });

    it('should specify trackFullReportClicked property on full report action', async () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

      createComponent();

      await waitForPromises();

      expect(findWidget().props('actionButtons')[0]).toMatchObject({
        trackFullReportClicked: true,
      });
    });

    it('hides the manage licenses button when URL is not available', async () => {
      mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);

      createComponent({
        license_scanning: {
          settings_path: '',
          full_report_path: fullReportPath,
        },
      });

      await waitForPromises();

      expect(findFullReportLink().exists()).toBe(true);
      expect(findFullReportLink().text()).toBe('Full report');

      expect(findActionButtons().exists()).toBe(true);
    });
  });

  describe('expanded data', () => {
    describe('with new licenses', () => {
      beforeEach(async () => {
        mockApi(licenseComparisonPathCollapsed, HTTP_STATUS_OK, licenseComplianceNewLicenses);
        mockApi(licenseComparisonPath, HTTP_STATUS_OK, licenseComplianceSuccessExpanded);

        createComponent();

        await waitForPromises();

        findToggleCollapsedButton().trigger('click');

        await waitForPromises();
      });

      it('displays denied licenses', () => {
        expect(findAllExtensionListItems().at(0).element).toMatchSnapshot();
      });

      it('displays uncategorized licenses', () => {
        expect(findAllExtensionListItems().at(1).element).toMatchSnapshot();
      });

      it('displays allowed licenses', () => {
        expect(findAllExtensionListItems().at(2).element).toMatchSnapshot();
      });
    });
  });
});
