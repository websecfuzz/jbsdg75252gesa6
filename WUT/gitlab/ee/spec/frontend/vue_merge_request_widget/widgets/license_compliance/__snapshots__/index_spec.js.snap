// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`License Compliance extension expanded data with new licenses displays allowed licenses 1`] = `
<div
  class="gl-border-t gl-border-t-section gl-flex gl-items-baseline gl-pl-7 gl-py-3"
  data-testid="extension-list-item"
>
  <div
    class="gl-min-w-0 gl-w-full"
  >
    <div
      class="gl-flex"
    >
      <div
        class="gl-mb-2"
      >
        <strong
          class="gl-block"
        >
          Allowed
        </strong>
      </div>
    </div>
    <div
      class="gl-flex gl-items-baseline"
    >
      <div
        class="gl-flex gl-flex-col gl-w-full"
      >
        <div
          class="gl-flex gl-grow"
        >
          <div
            class="gl-flex gl-grow gl-items-baseline"
          >
            <div>
              <p
                class="gl-mb-0 gl-mr-1"
              >
                Acceptable for use in this project
              </p>
            </div>
          </div>
        </div>
        <ul
          class="gl-list-none gl-m-0 gl-p-0"
        >
          <li>
            <div
              class="gl-flex gl-items-baseline"
            >
              <div
                class="gl-mr-3 gl-relative gl-rounded-full"
              >
                <svg
                  aria-label="Success WidgetLicenseCompliance"
                  class="gl-fill-current gl-fill-status-success gl-icon gl-relative gl-z-1 s12"
                  data-testid="status-success-icon"
                  role="img"
                >
                  <use
                    href="file-mock#status-success"
                  />
                </svg>
              </div>
              <div
                class="gl-min-w-0 gl-w-full"
              >
                <div
                  class="gl-flex gl-items-baseline"
                >
                  <div
                    class="gl-flex gl-flex-col gl-w-full"
                  >
                    <div
                      class="gl-flex gl-grow"
                    >
                      <div
                        class="gl-flex gl-grow gl-items-baseline"
                      >
                        <div>
                          <p
                            class="gl-mb-0 gl-mr-1"
                          />
                          <a
                            class="gl-link"
                            href="https://opensource.org/licenses/ISC"
                          >
                            ISC License
                          </a>
                        </div>
                      </div>
                      <div
                        class="gl-flex gl-items-start gl-ml-auto gl-pl-3"
                      >
                        <div
                          class="gl-block gl-disclosure-dropdown gl-new-dropdown md:!gl-hidden"
                        >
                          <button
                            aria-controls="reference-1"
                            aria-expanded="false"
                            class="!gl-p-2 btn btn-default btn-default-tertiary btn-icon btn-sm gl-button gl-new-dropdown-icon-only gl-new-dropdown-toggle gl-new-dropdown-toggle-no-caret"
                            data-testid="base-dropdown-toggle"
                            id="reference-0"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="gl-button-icon gl-fill-current gl-icon s16"
                              data-testid="ellipsis_v-icon"
                              role="img"
                            >
                              <use
                                href="file-mock#ellipsis_v"
                              />
                            </svg>
                            <span
                              class="gl-button-text"
                            >
                              <span
                                class="gl-new-dropdown-button-text gl-sr-only"
                              />
                            </span>
                          </button>
                          <div
                            class="!gl-w-31 gl-new-dropdown-panel"
                            data-testid="base-dropdown-menu"
                            id="reference-1"
                          >
                            <div
                              class="gl-new-dropdown-arrow"
                            />
                            <div
                              class="gl-new-dropdown-inner"
                            >
                              <ul
                                aria-labelledby="reference-0"
                                class="gl-new-dropdown-contents"
                                data-testid="disclosure-content"
                                id="reference-2"
                                tabindex="-1"
                              >
                                <li
                                  class="gl-new-dropdown-item"
                                  data-testid="disclosure-dropdown-item"
                                  tabindex="0"
                                >
                                  <a
                                    class="gl-new-dropdown-item-content"
                                    href="/group-name/project-name/-/merge_requests/78/full_report"
                                    tabindex="-1"
                                  >
                                    <span
                                      class="gl-new-dropdown-item-text-wrapper"
                                    >
                                      <span
                                        class="gl-flex gl-items-center gl-justify-between"
                                      >
                                        Used by 2 packages
                                      </span>
                                    </span>
                                  </a>
                                </li>
                              </ul>
                            </div>
                          </div>
                        </div>
                        <a
                          class="btn btn-confirm btn-confirm-tertiary btn-sm gl-button gl-float-left gl-hidden md:gl-inline-flex"
                          data-testid="extension-actions-button"
                          href="/group-name/project-name/-/merge_requests/78/full_report"
                        >
                          <span
                            class="gl-button-text"
                          >
                            Used by 2 packages
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`License Compliance extension expanded data with new licenses displays denied licenses 1`] = `
<div
  class="gl-border-t gl-border-t-0 gl-border-t-section gl-flex gl-items-baseline gl-pl-7 gl-py-3"
  data-testid="extension-list-item"
>
  <div
    class="gl-min-w-0 gl-w-full"
  >
    <div
      class="gl-flex"
    >
      <div
        class="gl-mb-2"
      >
        <strong
          class="gl-block"
        >
          Denied
        </strong>
      </div>
    </div>
    <div
      class="gl-flex gl-items-baseline"
    >
      <div
        class="gl-flex gl-flex-col gl-w-full"
      >
        <div
          class="gl-flex gl-grow"
        >
          <div
            class="gl-flex gl-grow gl-items-baseline"
          >
            <div>
              <p
                class="gl-mb-0 gl-mr-1"
              >
                Out-of-compliance with the project's policies and should be removed
              </p>
            </div>
          </div>
        </div>
        <ul
          class="gl-list-none gl-m-0 gl-p-0"
        >
          <li>
            <div
              class="gl-flex gl-items-start"
            >
              <div
                class="gl-mr-3 gl-relative gl-rounded-full"
              >
                <svg
                  aria-label="Failed WidgetLicenseCompliance"
                  class="gl-fill-current gl-fill-status-danger gl-icon gl-relative gl-z-1 s12"
                  data-testid="status-failed-icon"
                  role="img"
                >
                  <use
                    href="file-mock#status-failed"
                  />
                </svg>
              </div>
              <div
                class="gl-min-w-0 gl-w-full"
              >
                <div
                  class="gl-flex gl-items-baseline"
                >
                  <div
                    class="gl-flex gl-flex-col gl-w-full"
                  >
                    <div
                      class="gl-flex gl-grow"
                    >
                      <div
                        class="gl-flex gl-grow gl-items-baseline"
                      >
                        <div>
                          <p
                            class="gl-mb-0 gl-mr-1"
                          />
                          <a
                            class="gl-link"
                            href="https://opensource.org/licenses/Apache-2.0"
                          >
                            Apache License 2.0
                          </a>
                          <p
                            class="gl-mb-0"
                          >
                            Used by websocket-driver, websocket-extensions, xml-name-validator
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`License Compliance extension expanded data with new licenses displays uncategorized licenses 1`] = `
<div
  class="gl-border-t gl-border-t-section gl-flex gl-items-baseline gl-pl-7 gl-py-3"
  data-testid="extension-list-item"
>
  <div
    class="gl-min-w-0 gl-w-full"
  >
    <div
      class="gl-flex"
    >
      <div
        class="gl-mb-2"
      >
        <strong
          class="gl-block"
        >
          Uncategorized
        </strong>
      </div>
    </div>
    <div
      class="gl-flex gl-items-baseline"
    >
      <div
        class="gl-flex gl-flex-col gl-w-full"
      >
        <div
          class="gl-flex gl-grow"
        >
          <div
            class="gl-flex gl-grow gl-items-baseline"
          >
            <div>
              <p
                class="gl-mb-0 gl-mr-1"
              >
                No policy matches this license
              </p>
            </div>
          </div>
        </div>
        <ul
          class="gl-list-none gl-m-0 gl-p-0"
        >
          <li>
            <div
              class="gl-flex gl-items-baseline"
            >
              <div
                class="gl-mr-3 gl-relative gl-rounded-full"
              >
                <svg
                  aria-label="Notice WidgetLicenseCompliance"
                  class="gl-fill-current gl-fill-status-neutral gl-icon gl-relative gl-z-1 s12"
                  data-testid="status-notice-icon"
                  role="img"
                >
                  <use
                    href="file-mock#status-alert"
                  />
                </svg>
              </div>
              <div
                class="gl-min-w-0 gl-w-full"
              >
                <div
                  class="gl-flex gl-items-baseline"
                >
                  <div
                    class="gl-flex gl-flex-col gl-w-full"
                  >
                    <div
                      class="gl-flex gl-grow"
                    >
                      <div
                        class="gl-flex gl-grow gl-items-baseline"
                      >
                        <div>
                          <p
                            class="gl-mb-0 gl-mr-1"
                          />
                          <a
                            class="gl-link"
                            href="http://opensource.linux-mirror.org/licenses/afl-2.1.txt"
                          >
                            Academic Free License v2.1
                          </a>
                        </div>
                      </div>
                      <div
                        class="gl-flex gl-items-start gl-ml-auto gl-pl-3"
                      >
                        <div
                          class="gl-block gl-disclosure-dropdown gl-new-dropdown md:!gl-hidden"
                        >
                          <button
                            aria-controls="reference-1"
                            aria-expanded="false"
                            class="!gl-p-2 btn btn-default btn-default-tertiary btn-icon btn-sm gl-button gl-new-dropdown-icon-only gl-new-dropdown-toggle gl-new-dropdown-toggle-no-caret"
                            data-testid="base-dropdown-toggle"
                            id="reference-0"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="gl-button-icon gl-fill-current gl-icon s16"
                              data-testid="ellipsis_v-icon"
                              role="img"
                            >
                              <use
                                href="file-mock#ellipsis_v"
                              />
                            </svg>
                            <span
                              class="gl-button-text"
                            >
                              <span
                                class="gl-new-dropdown-button-text gl-sr-only"
                              />
                            </span>
                          </button>
                          <div
                            class="!gl-w-31 gl-new-dropdown-panel"
                            data-testid="base-dropdown-menu"
                            id="reference-1"
                          >
                            <div
                              class="gl-new-dropdown-arrow"
                            />
                            <div
                              class="gl-new-dropdown-inner"
                            >
                              <ul
                                aria-labelledby="reference-0"
                                class="gl-new-dropdown-contents"
                                data-testid="disclosure-content"
                                id="reference-2"
                                tabindex="-1"
                              >
                                <li
                                  class="gl-new-dropdown-item"
                                  data-testid="disclosure-dropdown-item"
                                  tabindex="0"
                                >
                                  <a
                                    class="gl-new-dropdown-item-content"
                                    href="/group-name/project-name/-/merge_requests/78/full_report"
                                    tabindex="-1"
                                  >
                                    <span
                                      class="gl-new-dropdown-item-text-wrapper"
                                    >
                                      <span
                                        class="gl-flex gl-items-center gl-justify-between"
                                      >
                                        Used by 1 package
                                      </span>
                                    </span>
                                  </a>
                                </li>
                              </ul>
                            </div>
                          </div>
                        </div>
                        <a
                          class="btn btn-confirm btn-confirm-tertiary btn-sm gl-button gl-float-left gl-hidden md:gl-inline-flex"
                          data-testid="extension-actions-button"
                          href="/group-name/project-name/-/merge_requests/78/full_report"
                        >
                          <span
                            class="gl-button-text"
                          >
                            Used by 1 package
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
